src/common/dao/notification/notification_job.go

```diff
@@ -0,0 +1,139 @@
+package notification
+
+import (
+	"fmt"
+
+	"github.com/astaxie/beego/orm"
+	"github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/pkg/errors"
+)
+
+// UpdateNotificationJob update notification job
+func UpdateNotificationJob(job *models.NotificationJob, props ...string) (int64, error) {
+	if job == nil {
+		return 0, errors.New("nil job")
+	}
+
+	if job.ID == 0 {
+		return 0, fmt.Errorf("notification job ID is empty")
+	}
+
+	o := dao.GetOrmer()
+	return o.Update(job, props...)
+}
+
+// UpdateNotificationJobStatus ...
+func UpdateNotificationJobStatus(id int64, status string, statusCondition ...string) (int64, error) {
```

Comment:
1. Please add comment to clarify the usage of statusCondition, what will happen if it's not met.
2. Is it correct to only allow one statusCondition, for example think of the state transition, multiple status may transfer to "error"?

Reply:
> 1. Please add comment to clarify the usage of statusCondition, what will happen if it's not met.
> 2. Is it correct to only allow one statusCondition, for example think of the state transition, multiple status may transfer to "error"?

1. comments will be add;
2. will change the code to support multiple status(original) transformation.